File: w2d1/chess_javascript/src/null_piece.js
import Piece from './piece'

class NullPiece extends Piece {
	constructor() {
		super();
		this.color = null;
	}

	toString() {
		return " ";
	}
}

let p = new NullPiece;
console.log(p.color);

export default Piece;

File: w2d1/chess_python/Pieces/null_piece.py
from piece import Piece

class NullPiece(Piece):
	def __init__(self, board, color, pos):
		Piece.__init__(self, board, color, pos)

	def to_s(self):
		return '   '

	def valid_moves(self):
		raise Exception('No piece at ' + str(self.pos))
